core-web/libs/edit-content/src/lib/utils/functions.util.ts

```diff
@@ -141,15 +141,22 @@ export const isValidJson = (value: string): boolean => {
  * @param {DotCMSContentTypeFieldVariable[]} fieldVariables - The array of field variables to be parsed.
  * @return {T} - The parsed object with key-value pairs.
  */
-export const getFieldVariablesParsed = <T extends Record<string, string>>(
+export const getFieldVariablesParsed = <T extends Record<string, string | boolean>>(
     fieldVariables: DotCMSContentTypeFieldVariable[]
 ): T => {
     if (!fieldVariables) {
         return {} as T;
     }
 
-    const result: Record<string, string> = {};
+    const result = {};
     fieldVariables.forEach(({ key, value }) => {
+        // If the value is a boolean string, convert it to a boolean
+        if (value === 'true' || value === 'false') {
+            result[key] = value === 'true';
```

Comment:
just to be safe with this check, make a `.toLowercase()`

Reply:
We should not modify the variable to compare it. Here are my reasons:

1. This method is generic.
2. Users can use those variables (Custom Fields). If they need the string "True" or "TRUE" instead of the boolean that will be an `edge case`. 

If they make a mistake when writing a variable value, they can correct it.

Reply:
I would to know what you think @fmontes @zJaaal @KevinDavilaDotCMS ?

Reply:
I agree with your reasons